### PR TITLE
[substrate] #INFRA-31 Add ipfs helm chart for vitalam-ipfs-node

### PR DESCRIPTION
**Pull request by TonyRowntree**
Signed-off-by: TonyRowntree <33454202+TonyRowntree@users.noreply.github.com>

**Changelog**
- Added the IPFS helm charts for vitalam-ipfs-node.

 

**Reviewed by**

 

**Linked issue**
#INFRA-31

### CODE REVIEW

File: platforms/substrate/charts/vitalam-ipfs-node/ci/ct-values.yaml [hunk 1]
```diff
@@ -0,0 +1,17 @@
+image:
```
Comment (sownak) [inline platforms/substrate/charts/vitalam-ipfs-node/ci/ct-values.yaml]: dont think ci is needed to be copied

File: platforms/substrate/charts/vitalam-ipfs-node/Chart.yaml [hunk 2]
```diff
@@ -0,0 +1,16 @@
+apiVersion: v2
+name: vitalam-ipfs
```
Comment (sownak) [inline platforms/substrate/charts/vitalam-ipfs-node/Chart.yaml]: chart name should be same as folder name.

File: platforms/substrate/charts/vitalam-ipfs-node/templates/statefulset.yaml [hunk 3]
```diff
@@ -0,0 +1,116 @@
...
+          command: ['/bin/sh', '-c', '-e', '-x']
+          args:
+            - |
+              [ ! -f "$IPFS_PATH/config" ] && $ipfsCommand init --profile server;
+              $ipfsCommand config Addresses.API /ip4/0.0.0.0/tcp/$ipfsApiPort;
+              $ipfsCommand config --json Addresses.Swarm "[\"/ip4/0.0.0.0/tcp/$ipfsSwarmPort\"]";
+              $ipfsCommand config Datastore.StorageMax $(echo "$dataVolumeSize""Gb");
+              $ipfsCommand config --json API.HTTPHeaders.Access-Control-Allow-Origin "[\"http://0.0.0.0:$ipfsApiPort\", \"http://127.0.0.1:$ipfsApiPort\"]";
+              $ipfsCommand bootstrap rm --all;
+              if [ ! -z "$ipfsBootNodeAddress" ]; then $ipfsCommand bootstrap add $ipfsBootNodeAddress; fi;
```
Comment (sownak) [inline platforms/substrate/charts/vitalam-ipfs-node/templates/statefulset.yaml:40]: These may need to be changed for our deployment as this is used localhost as IPs

File: platforms/substrate/charts/vitalam-ipfs-node/templates/statefulset.yaml [hunk 4]
```diff
@@ -0,0 +1,116 @@
...
+      metadata:
+        name: ipfs-data
+      spec:
+        accessModes: [ "ReadWriteOnce" ]
+        {{- if .Values.storage.storageClass }}
+        storageClassName: {{ .Values.storage.storageClass }}
+        {{- end }}
+        resources:
+          requests:
+            storage: {{ printf "%vGi" .Values.storage.dataVolumeSize | quote }}
```
Comment (sownak) [inline platforms/substrate/charts/vitalam-ipfs-node/templates/statefulset.yaml]: newline missing

File: platforms/substrate/charts/vitalam-ipfs-node/values.yaml [hunk 5]
```diff
@@ -0,0 +1,27 @@
+# fullNameOverride:
```
Comment (sownak) [inline platforms/substrate/charts/vitalam-ipfs-node/values.yaml:1]: Add more comments and examples in values.yaml like other charts in Bevel

File: platforms/substrate/charts/vitalam-ipfs-node/Chart.yaml [hunk 6]
```diff
@@ -0,0 +1,16 @@
...
+dependencies:
+  - name: vitalam-node
+    version: '0.23.5'
+    repository: https://digicatapult.github.io/vitalam-node/
+    condition: vitalamNode.enabled
+    alias: vitalamNode
+maintainers:
+  - name: digicatapult
+    email: opensource@digicatapult.org.uk
+    url: www.digicatapult.org.uk
```
Comment (sownak) [inline platforms/substrate/charts/vitalam-ipfs-node/Chart.yaml]: Instead of this, add a comment and link to digitalcatapult's original repo

**Review by sownak — CHANGES_REQUESTED** [review]

File: platforms/substrate/charts/vitalam-ipfs-node/templates/statefulset.yaml (hunk 3 above)
Comment (TonyRowntree) [reply platforms/substrate/charts/vitalam-ipfs-node/templates/statefulset.yaml:40]: @sownak what would the values be for our deployment?

Comment (sownak) [reply platforms/substrate/charts/vitalam-ipfs-node/templates/statefulset.yaml:40]: These can remain for now, you will need to update as part of your next task.

**Review by sownak — APPROVED** [review]

**Pull request merged**